zmq/tests/test_crash.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (C) PyZMQ Developers
+# Distributed under the terms of the Modified BSD License.
+
+
+import functools
+import hashlib
+from multiprocessing import Pipe, Process
+from random import Random
+import time
+
+import pytest
+
+import zmq
+
+from zmq.tests import BaseZMQTestCase, GreenTest, have_gevent
+
+
+def topic(x):
+    """Generates a PUB/SUB topic from a number."""
+    return hashlib.md5(str(x)).hexdigest()[:8]
```

Comment:
A simpler way to create a topic from a number is to use the number itself: `return struct.pack('Q', x)`

Reply:
The topics should not share a common prefix.  Hashing is necessary to reproduce this issue.

Reply:
That's not accurate. This failure is reproducible using struct.pack. They may have a common prefix, but the topics are still at least as unique (perfectly unique in the struct.pack case, almost certainly unique in the md5 case). The same is true of md5, where 1/16 start with 'a', etc.